piptools/repositories/pypi.py

```diff
@@ -62,6 +63,8 @@ def __init__(self, pip_args, cache_dir, build_isolation=False):
         # and pre) are deferred to pip.
         self.command = create_install_command()
         self.options, _ = self.command.parse_args(pip_args)
+        if self.options.cache_dir:
+            self.options.cache_dir = normalize_path(self.options.cache_dir)
```

Comment:
Just wondering why pip doesn't normalize `options.cache_dir` at [the parse-options level](https://github.com/pypa/pip/blob/723b84517d54ee37a6a43dd5d8d30cbf3525b0c4/src/pip/_internal/cli/cmdoptions.py#L640-L648)?

Reply:
Softly ping @pradyunsg. Do you have any insights?

Reply:
Off the top of my head, because we didn't come around to refactoring that bit of code yet.

It'd probably help to take a look at the PR that added the normalisation, to make sure if my intuition is correct though. :)

Reply:
@pradyunsg thanks! Should we raise an issue on pip's issue tracker?

Reply:
I think we'll fix it eventually, even if there's no tracking issue. Doesn't mean that having a tracking issue would hurt tho. ;)